website/server/libs/payments/subscriptions.js

```diff
@@ -79,6 +80,20 @@ async function createSubscription (data) {
   let emailType = 'subscription-begins';
   let recipientIsSubscribed = recipient.isSubscribed();
 
+  if (data.user && !data.gift && !data.groupId) {
+    const unlockedUser = await User.findOneAndUpdate(
+      {
+        _id: data.user._id,
+        $or: [
+          { _subSignature: 'NOT_RUNNING' },
+          { _subSignature: { $exists: false } },
+        ],
+      },
+      { $set: { _subSignature: 'SUB_IN_PROGRESS' } },
+    );
+    if (!unlockedUser) throw new NotFound('User not found or subscription already processing.');
```

Comment:
better use `{}` so many single line if conditions we might loose the real sight of what will/should be executed when - I think I should search for a lint for it xD